src/item.cpp

```diff
@@ -2523,6 +2523,21 @@ long item::get_property_long( const std::string& prop, long def ) const
     return def;
 }
 
+int item::get_quality(const std::string &quality_id) const
+{
+    int return_quality = INT_MIN;
+    for( const auto &quality : type->qualities ) {
```

Comment:
Instead of looping, you can just find the entry in the map:
```C++
const auto iter = type->quality.find( id );
if( iter != type->quality ) { // found it!
    return_quality = iter->second;
}
```